src/components/Debug.ts

```diff
@@ -0,0 +1,159 @@
+ * executing flux. It is meant to be a temporary solution to allow F5-to-run abilities,
+ * but does not provide any actual debugging. This interface will eventually be replaced
+ * by a wasm server with more capabilities.
+ */
+class DebugSession extends LoggingDebugSession {
+    private configurationDone = new Subject()
+
+    public constructor(private context : vscode.ExtensionContext) {
+        super('')
+    }
+
+    /**
+     * The 'initialize' request is the first request called by the frontend
+     * to interrogate the features the debug adapter provides.
+     */
+    protected initializeRequest(response : DebugProtocol.InitializeResponse, _args : DebugProtocol.InitializeRequestArguments) : void {
+        response.body = response.body || {}
+        response.body.supportsConfigurationDoneRequest = true
+        response.body.supportsEvaluateForHovers = false
+        response.body.supportsStepBack = false
+        response.body.supportsDataBreakpoints = false
+        response.body.supportsCompletionsRequest = false
+        response.body.completionTriggerCharacters = []
+        response.body.supportsCancelRequest = false
+        response.body.supportsBreakpointLocationsRequest = false
+        response.body.supportsStepInTargetsRequest = false
+        response.body.supportsExceptionFilterOptions = false
+        response.body.exceptionBreakpointFilters = []
+
+        // make VS Code send exceptionInfo request
+        response.body.supportsExceptionInfoRequest = false
+
+        // make VS Code send setVariable request
+        response.body.supportsSetVariable = false
+
+        // make VS Code send setExpression request
+        response.body.supportsSetExpression = false
+
+        // make VS Code send disassemble request
+        response.body.supportsDisassembleRequest = false
+        response.body.supportsSteppingGranularity = false
+        response.body.supportsInstructionBreakpoints = false
+
+        this.sendResponse(response)
+
+        // Notify the client that the initialization is complete, which signals the ability to accept
+        // configuration requests at any time. The client will signal the end of its configuration sequence by
+        // sending a `configurationDone` request.
+        this.sendEvent(new InitializedEvent())
+    }
+
+    /**
+     * Called at the end of the configuration sequence.
+     * Indicates that all breakpoints etc. have been sent to the DA and that the 'launch' can start.
+     */
+    protected configurationDoneRequest(response : DebugProtocol.ConfigurationDoneResponse, args : DebugProtocol.ConfigurationDoneArguments) : void {
+        super.configurationDoneRequest(response, args)
+
+        // notify the launchRequest that configuration has finished
+        this.configurationDone.notify()
+    }
+
+    protected async launchRequest(response : DebugProtocol.LaunchResponse, args : ILaunchRequestArguments) : Promise<void> {
+        console.log('launchRequest')
```

Comment:
I thought I had grabbed them all, but I guess not. This comment drove me to add a lint rule to prevent it from passing CI with `console.log` in the future.